### PR TITLE
Myfix

### PR DESCRIPTION
**Pull request by ffilz**
Fix Coverity NULL dereference errors

### CODE REVIEW

File: src/auth_unix.c [hunk 1]
```diff
@@ -146,13 +145,10 @@ authunix_ncreate(char *machname, uid_t uid, gid_t gid, int len,
...
 cleanup_authunix_create:
-	if (auth)
-		mem_free(auth, sizeof(*auth));
-	if (au) {
-		if (au->au_origcred.oa_base)
-			mem_free(au->au_origcred.oa_base, (size_t) len);
-		mem_free(au, sizeof(*au));
-	}
+	mem_free(auth, sizeof(*auth));
+	if (au->au_origcred.oa_base)
```
Comment (was4) [inline src/auth_unix.c]: I don't see how this "if" will ever be true. It is only set after the goto above.

File: src/svc_dg.c [hunk 2]
```diff
@@ -182,12 +182,9 @@ svc_dg_ncreate(int fd, u_int sendsize, u_int recvsize)
...
  freedata:
 	__warnx(TIRPC_DEBUG_FLAG_SVC_DG, svc_dg_str, __no_mem_str);
-	if (xprt) {
-		if (su)
-			mem_free(su, sizeof(*su));
-		xprt_unregister(xprt);
-		mem_free(xprt, sizeof(SVCXPRT));
-	}
+	mem_free(su, sizeof(*su));
+	xprt_unregister(xprt);
```
Comment (was4) [inline src/svc_dg.c:186]: I don't see how this could have been registered. The only call to freedata is before the register.

File: src/svc_vc.c [hunk 3]
```diff
@@ -249,8 +249,7 @@ svc_vc_ncreate2(int fd, u_int sendsize, u_int recvsize, u_int flags)
 	return (xprt);
 
  err:
-	if (rdvs)
-		mem_free(rdvs, sizeof(struct cf_rendezvous));
+	mem_free(rdvs, sizeof(struct cf_rendezvous));
```
Comment (was4) [inline src/svc_vc.c:252]: You can remove the "= NULL" in the initializer, too.

File: src/auth_unix.c (hunk 1 above)
Comment (was4) [inline src/auth_unix.c]: I don't see how the "if" can ever be true. The only goto is before oa_base is set.

File: src/svc_dg.c (hunk 2 above)
Comment (ffilz) [reply src/svc_dg.c:186]: Hmm, that's probably true, but I'm not sure I should clean that up in this patch

File: src/svc_vc.c (hunk 3 above)
Comment (ffilz) [reply src/svc_vc.c:252]: Updated the patch (also caught one more place while poking around).

**Comment by mattbenjamin** [comment]
merged by hand

**Pull request closed without merging**